pkg/util/cluster/aad.go

```diff
@@ -66,11 +50,12 @@ func (c *Cluster) createServicePrincipal(ctx context.Context, appID string) (str
 	// wait.PollImmediateUntil. Doing this will not propagate the latest error
 	// to the user in case when wait exceeds the timeout
 	_ = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
-		sp, err = c.serviceprincipals.Create(ctx, azgraphrbac.ServicePrincipalCreateParameters{
-			AppID: &appID,
-		})
-		if detailedErr, ok := err.(autorest.DetailedError); ok &&
-			detailedErr.StatusCode == http.StatusForbidden {
+		requestBody := msgraph_models.NewServicePrincipal()
+		requestBody.SetAppId(&appID)
+		result, err = c.graph.ServicePrincipals().Post(ctx, requestBody, nil)
+
+		if oDataError, ok := err.(msgraph_errors.ODataErrorable); ok &&
+			*oDataError.GetError().GetCode() == "accessDenied" {
```

Comment:
can we still check for a 403 here, or is checking the "errorCode" the recommended check? The comments below indicate the code will be "Unknown" if this is no longer the case please update the comment as well.

Reply:
The SDK seems to want you to cast errors to the [Error resource type](https://learn.microsoft.com/en-us/graph/errors#error-resource-type).  I assume raw HTTP error codes can be extracted somehow but are probably no easier to get to.

The [accessDenied](https://learn.microsoft.com/en-us/graph/errors#code-property) code for MS Graph requests seems to be the closest equivalent to **Authorization_RequestDenied** from AD Graph, but I haven't actually reproduced this error so I'm hesitant to update the comment until we see what the exact MS Graph error looks like.

Reply:
Yeah, I don't even know _how_ to reproduce this outside of a production environment.  Service principals are already established in our development and integration tenants so this should never occur there.  As usual, Jim's [original commit message](https://github.com/Azure/ARO-RP/commit/33eb07725c90ea9deb6e6f55a2325f9d677091b4) doesn't offer any additional context.  Open to suggestions if you know of a way.

Reply:
Would an error ever be something other than `accessDenied`, should we have an `else err != nil` to handle that as well?

Reply:
I think that's covered below with `return err == nil, err`.